04-onion-routing.md

```diff
@@ -873,8 +931,54 @@ An _intermediate hop_ MUST NOT, but the _final node_:
   the final node's HTLC:
     - MUST return `final_incorrect_cltv_expiry` error.
   - if the `amt_to_forward` is greater than the `incoming_htlc_amt` from the
-  final node's HTLC:
+  final node's HTLC, AND `incomplete_payment` is not set:
     - MUST return a `final_incorrect_htlc_amount` error.
+  - if `incomplete_payment` is set:
+    - MUST use `amt_to_forward` to evaluate for
+      `incorrect_payment_amount` errors.
+    - SHOULD suspend incoming HTLCs:
```

Comment:
Maybe change to : SHOULD suspend incoming HTLCs with the same payment hash:

Reply:
My mental model is a single-threaded server, so approximately, you would write code as below:

1.  If the `incomplete_payment` is set, suspend incoming HTLC (also set a timeout trigger to fail suspended HTLCs due to taking too long)
2.  Scan through all suspended HTLCs and sum up their total.
3.  If the total in all suspended HTLCs >= `amt_to_forward`, then claim them.

Although the changed wording seems fine to me, either way.